### PR TITLE
Re-engineer Domain and Resource tests in response to #220

### PR DESCRIPTION
**Pull request by GeraldScott**
This pull request is to resolve Issue #220 

The entity resource tests were failing because they were using the size of the array returned by the .get("/api/<%= entityApiUrl %>") clause as a proxy for the size of the dataset in the database.

There are a number of other problems with the REST tests:
- they are very slow because they are being used to test for required fields
- they are effectively Integration tests, not Unit tests because they use APIs to set up data and to test for database activity
- they are testing JPA functionality instead of leaving that to Hibernate and Liquibase

This Pull Request does the following:
- the REST tests have been stripped down to test for basic Create/Read/Update/Delete 
- database size before and after updates is checked using built-in Panache functionality, not an API call (which was causing the problem in the first place)
- the Domain tests have been enhanced to check for JPA annotations to test for "required" fields (instead of doing an API call as is curently the case) 
- there are two new test helper classes that implement the reflection to test for the JPA annotations
- add another domain test to check for "unique" fields

The advantages are:
- Issue 230 is resolved
- REST tests are much faster because they only use the API calls that are under test
- the domain tests of JPA annotations are more useful
- the test helper classes can be extended to test for any JPA annotation (@Min, @Max, etc), including the value of the annotation ("max = 256" for example)

### CODE REVIEW

File: generators/entity-server/templates/quarkus/src/test/java/package/_partials/create.ejs [hunk 1]
```diff
@@ -18,20 +18,8 @@
 -%>
     @Test
     public void create<%= entityClass %>() {
```
Comment (danielpetisme) [inline generators/entity-server/templates/quarkus/src/test/java/package/_partials/create.ejs:20]: Please keep the validation tests (API should failed when creating with an existing ID)

File: generators/entity-server/templates/quarkus/src/test/java/package/_partials/create_with_field_required.ejs [hunk 2]
```diff
@@ -1,52 +0,0 @@
-<%_ for (let idx = 0; idx < fields.length; idx++) {
```
Comment (danielpetisme) [inline generators/entity-server/templates/quarkus/src/test/java/package/_partials/create_with_field_required.ejs:1]: Please keep the validation tests (fields required)

File: generators/entity-server/templates/quarkus/src/test/java/package/_partials/get_all.ejs [hunk 3]
```diff
@@ -31,8 +34,21 @@
             .then()
             .statusCode(CREATED.getStatusCode())
             .extract().as(ENTITY_TYPE);
+    <%_ if (hasPagination) { _%>
 
-        // Get all the <%= entityInstance %>List
+        // Compare the number of records returned with the number at the start of the test
+        assertThat(Integer.parseInt(
```
Comment (danielpetisme) [inline generators/entity-server/templates/quarkus/src/test/java/package/_partials/get_all.ejs:40]: try to keep assertation as "fluent" as possible (ie. assertation should look like be a plain English sentence).
Could you please extract the RESTAssured call into a local variable?

File: generators/entity-server/templates/quarkus/src/test/java/package/domain/EntityTest.java.ejs [hunk 4]
```diff
@@ -54,4 +67,43 @@ public class <%= entityClass %>Test {
         assertThat(<%= asEntity(entityInstance) %>1).isNotEqualTo(<%= asEntity(entityInstance) %>2);
     <%_} _%>
     }
+    
+    @Test
+    public void entityAnnotations() throws Exception {
```
Comment (danielpetisme) [inline generators/entity-server/templates/quarkus/src/test/java/package/domain/EntityTest.java.ejs:72]: What's the purpose of this test?
Sounds like a test `ArchTest.java`
I think this kind of "technical testing" should be done in a dedicated class. 
EntityTest should be dedicated to validate common entities behaviors, not their definition.

File: generators/entity-server/templates/quarkus/src/test/java/package/web/rest/EntityResourceTest.java.ejs [hunk 5]
```diff
@@ -102,17 +89,16 @@ public class <%= entityClass%>ResourceTest {
...
     }
 
+<%- include('../../_partials/get_all', this); -%>
+
+<%- include('../../_partials/get', this); -%>
     <%_ if (!readOnly) { _%>
+    
 <%- include('../../_partials/create', this); -%>
 
-<%- include('../../_partials/create_with_field_required', this); -%>
```
Comment (danielpetisme) [inline generators/entity-server/templates/quarkus/src/test/java/package/web/rest/EntityResourceTest.java.ejs:108]: Please re-include these tests

File: generators/server/templates/quarkus/src/test/java/package/ArchTest.java.ejs [hunk 6]
```diff
@@ -31,7 +31,7 @@ class ArchTest {
     void servicesAndRepositoriesShouldNotDependOnWebLayer() {
         JavaClasses importedClasses = new ClassFileImporter()
             .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
-            .importPackages("com.mycompany.myapp");
+            .importPackages("<%=packageName%>");
```
Comment (danielpetisme) [inline generators/server/templates/quarkus/src/test/java/package/ArchTest.java.ejs:34]: Very nice catch :)

File: generators/server/templates/quarkus/src/test/java/package/AssertAnnotations.java.ejs [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package <%=packageName%>;
+
+import java.util.Arrays;
+
+@SuppressWarnings("rawtypes")
+public class AssertAnnotations {
```
Comment (danielpetisme) [inline generators/server/templates/quarkus/src/test/java/package/AssertAnnotations.java.ejs:6]: I'm not very familiar with this kind of test.
Could you please elaborate your intent here

**Review by danielpetisme — CHANGES_REQUESTED** [review]
Hi @GeraldScott, first of all, THANK YOU! Thanks for contributing and for your amazing work done!
I had couple of comments. The 2 main concerns are:
- the validation tests (in the create Entity and the fields required testing)
- the technical tests on the annotation presence, I never saw that before, could you please detail the objective.

Looking forward to merging your work soon!

**Comment by mraible** [comment]
@GeraldScott Can you please rebase this PR to fix conflicts?

**Comment by GeraldScott** [comment]
Hi Colleagues

I would like to close this pull request. My attempts to rebase have been unsuccessful because too many dependency conflicts have emerged in the passage of time.

**Pull request closed without merging**